Hazel/Dtls/DtlsConnectionListener.cs

```diff
@@ -120,6 +120,7 @@ public void ResetPeer(ConnectionId connectionId, ulong nextExpectedSequenceNumbe
 
                 this.NextEpoch.State = HandshakeState.ExpectingHello;
                 this.NextEpoch.RecordProtection = null;
+                this.NextEpoch.Handshake = null;
```

Comment:
A little thing, shouldn't be a big deal either way, but above we call dispose, this handshake is disposed. We null or replace other stuff, but not this.